### PR TITLE
Memallocator

### PR DESCRIPTION
**Pull request by laura-pircalaboiu**

### CODE REVIEW

File: CMakeCache.txt [hunk 1]
```diff
@@ -0,0 +1,368 @@
+# This is the CMakeCache file.
```
Comment (NULLx76) [inline CMakeCache.txt:1]: This file probably needs to be removed?

File: kernel/src/vm/include/pmm.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/// This page allocator was completely designed and implemented by
+/// This page oldlocator was completely designed and implemented by
```
Comment (NULLx76) [inline kernel/src/vm/include/pmm.h:1]: Probably a rename gone wrong?

File: kernel/src/test/generate_tests.sh [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
+exit 0
```
Comment (NULLx76) [inline kernel/src/test/generate_tests.sh:9]: Is this supposed to be here?

File: kernel/src/ds/bpf/test/hashmap_test.c [hunk 4]
```diff
@@ -20,30 +20,30 @@ void simpleFree(void * val) {
...
-    int key = 5;
-    int data = 42;
-
-    hashmap__add(hm, &key, &data);
-
-    ASSERT(hashmap__find(hm, &key, NULL));
-
-    hashmap__free(hm);
+//    HashMap * hm = hashmap__new(int_hash_fn, int_compare_fn, fakeFree, fakeFree, NULL);
+//
```
Comment (NULLx76) [inline kernel/src/ds/bpf/test/hashmap_test.c:24]: These tests should probably be uncommented, or fixed if they are  faulty.

File: kernel/src/allocator/test/test_allocator.c [hunk 5]
```diff
@@ -1,52 +1,7 @@
...
-    uint8_t prev = *yolo;
-    *yolo = 42;
-    ASSERT_EQ(*yolo, 42);
-    *yolo = prev;
-
-    contract(heap);
-
-    ASSERT_EQ(heap->end, initial_end);
+TEST_CREATE(test_free_null, {
+    kfree(NULL);
```
Comment (NULLx76) [inline kernel/src/allocator/test/test_allocator.c:6]: We do probably want a couple more tests?

File: kernel/Makefile [hunk 6]
```diff
@@ -12,6 +12,7 @@ MEMORY = 1G
 LOG_LEVEL ?= 4
 DEFINITIONS = MEM_DEBUG LOG_LEVEL=${LOG_LEVEL}
 test: DEFINITIONS += ENABLE_TESTS # if we execute the test: rule, enable tests before recompiling
+DEFINITIONS += ENABLE_TESTS
```
Comment (NULLx76) [inline kernel/Makefile:15]: Debug line that probably needs to go away

File: course_os.cbp [hunk 7]
```diff
@@ -0,0 +1,269 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (NULLx76) [inline course_os.cbp:1]: This together with all(?) other CMakeFiles likely needs to go.

File: CMakeCache.txt (hunk 1 above)
Comment (jonay2000) [reply CMakeCache.txt]: fixed

File: kernel/src/test/generate_tests.sh (hunk 3 above)
Comment (jonay2000) [reply kernel/src/test/generate_tests.sh]: fixed

File: kernel/src/vm/include/pmm.h (hunk 2 above)
Comment (jonay2000) [reply kernel/src/vm/include/pmm.h]: fixed

File: kernel/src/ds/bpf/test/hashmap_test.c (hunk 4 above)
Comment (jonay2000) [reply kernel/src/ds/bpf/test/hashmap_test.c]: fixed

File: kernel/Makefile (hunk 6 above)
Comment (jonay2000) [reply kernel/Makefile:15]: fixed

File: course_os.cbp (hunk 7 above)
Comment (jonay2000) [reply course_os.cbp:1]: fixed

File: kernel/src/vm/include/asid_allocator.h [hunk 8]
```diff
@@ -4,7 +4,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-/// We have made a special allocator to keep track of all the ASIDs used and to determine if we
+/// We have made a special oldlocator to keep track of all the ASIDs used and to determine if we
```
Comment (NULLx76) [inline kernel/src/vm/include/asid_allocator.h:7]: Just one rename error renaming after that I think we can merge

**Pull request closed without merging**